packages/legend-query-builder/src/components/fetch-structure/QueryBuilderResultModifierPanel.tsx

```diff
@@ -58,20 +70,26 @@ const ColumnSortEditor = observer(
       label: sortState.columnState.columnName,
       value: sortState,
     };
+    const sortType = sortState.sortType;
+
     const onChange = (
       val: { label: string; value: QueryBuilderTDSColumnState } | null,
     ): void => {
       if (val !== null) {
         sortState.setColumnState(val.value);
       }
     };
-    const sortType = sortState.sortType;
 
-    const deleteColumnSort = (): void =>
-      tdsState.resultSetModifierState.deleteSortColumn(sortState);
+    const deleteColumnSort = (): void => {
+      const newSortColumns = [...sortColumns];
+      deleteEntry(newSortColumns, sortState);
```

Comment:
may I know the reason why we want to do it in this way instead of the original approach?

Reply:
The original approach would apply your changes to the state as soon as you made a change in the modal, so there was no good way to cancel your changes. This way, all the changes are stored locally within the modal first and then only applied to the state if you click Apply.

Since the modal has a local state storing the sortColumns array, we can't just mutate that array. We need to replace the whole array in the modal's state so that the component correctly re-renders when we add/remove sort columns.